### PR TITLE
16918 add BusinessLookupComponent

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#16918

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: package.json
Comment (severinbeauvais) [inline package.json:20]: Please do a `npm i` and also commit the package-lock file.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:145]: `inProgressBusinessLookup` is not declared in this component?

Comment (severinbeauvais) [reply src/components/new-request/search.vue:145]: OK, I found it in common-mixin.

File: src/mixins/common-mixin.ts
Comment (severinbeauvais) [inline src/mixins/common-mixin.ts:10]: This local property has no reason to exist in this mixin. It should be in the search component, where it can be initialized/reset when needed, etc.

**Comment by severinbeauvais** [comment]
I don't see the following:
- error handling (valid/invalid state)
- the selected business assigned to a variable (for future tickets to use)

Do you have screenshots to show?

File: src/services/business-lookup-services.ts
Comment (severinbeauvais) [inline src/services/business-lookup-services.ts:37]: What do you mean?

File: .env.example
Comment (severinbeauvais) [inline .env.example:24]: Looks good. Please also update ~/devops/vaults.env (which is used to populate the env keys from the 1Password vaults).

File: src/mixins/common-mixin.ts
Comment (kzdev420) [reply src/mixins/common-mixin.ts:10]: okay

File: .env.example
Comment (kzdev420) [reply .env.example:24]: I see

File: package.json
Comment (kzdev420) [reply package.json]: I did

File: .env.example
Comment (kzdev420) [reply .env.example:24]: I did

File: package.json
Comment (severinbeauvais) [inline package.json:1]: Remember to update the app version before we merge this.

Comment (kzdev420) [reply package.json:1]: Yes

File: src/components/new-request/business-lookup.vue
Comment (severinbeauvais) [inline src/components/new-request/business-lookup.vue:66]: Is this the component from Auth Web?

Does getting the Business Module here work?

File: src/components/new-request/business-lookup.vue
Comment (severinbeauvais) [inline src/components/new-request/business-lookup.vue:109]: This functionality is specific to Auth Web. I think you can remove it for Namerequest UI.

File: src/components/new-request/business-lookup.vue
Comment (severinbeauvais) [inline src/components/new-request/business-lookup.vue:99]: Using the Debounce decorator here is OK for now since this is Namerequest, but eventually we will want to change it to the lodash debounce function or something similar. You don't have to change this now, but now you know about it.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:177]: This is OK for now, but eventually (next ticket) we will want to store the entire event object, since we will need to look at the legal type also.

The code below to display the business name and id is also OK for now, just to demonstrate that business search works. Again, the next ticket(s) will remove this and actually do something with the business name/ legal type.

File: src/components/new-request/business-lookup.vue
Comment (severinbeauvais) [inline src/components/new-request/business-lookup.vue:104]: Set this so that we only search for active businesses.

**Comment by severinbeauvais** [comment]
Looks good. Can you show a couple of screenshots, please?

Also a note to devs who want to test this: there is a special key (which we won't post in GH) for accessing the business search API without a JWT (auth token). This is so we can search for a business before the user logs in. This will be sorted out in a different ticket.

**Comment by pwei1018** [comment]
Temporary Url for review: https://namerequest-dev--pr-638-62lfcpe2.web.app

**Comment by JazzarKarim** [comment]
> Also a note to devs who want to test this: there is a special key (which we won't post in GH) for accessing the business search API without a JWT (auth token). This is so we can search for a business before the user logs in. This will be sorted out in a different ticket.

I'm trying to use it in the temporary URL Sev however I'm seeing an error with the Content Security Policy.
dev error kevin

I've encountered that when I implemented Maximus' Webchat. It would run on my local but not in DEV. I had to add to the CSP an additional link. Could this also be the case here?

**Comment by JazzarKarim** [comment]
> > Also a note to devs who want to test this: there is a special key (which we won't post in GH) for accessing the business search API without a JWT (auth token). This is so we can search for a business before the user logs in. This will be sorted out in a different ticket.
> 
> I'm trying to use it in the temporary URL however I'm seeing an error with the Content Security Policy. dev error kevin
> 
> I've encountered that when I implemented Maximus' Webchat. It would run on my local but not in DEV. I had to add to the CSP an additional link. Could this also be the case here?

**Comment by severinbeauvais** [comment]
> I've encountered that when I implemented Maximus' Webchat. It would run on my local but not in DEV. I had to add to the CSP an additional link. Could this also be the case here?

Yeah, maybe. Try it and report back if it doesn't work.

File: src/components/new-request/business-lookup.vue
Comment (kzdev420) [reply src/components/new-request/business-lookup.vue:66]: Yes

File: src/components/new-request/business-lookup.vue
Comment (kzdev420) [reply src/components/new-request/business-lookup.vue:109]: Yes, I did

File: src/components/new-request/business-lookup.vue
Comment (severinbeauvais) [reply src/components/new-request/business-lookup.vue]: Thanks, resolved.

File: src/components/new-request/business-lookup.vue
Comment (severinbeauvais) [reply src/components/new-request/business-lookup.vue]: Thanks, resolved.

File: src/components/new-request/business-lookup.vue
Comment (severinbeauvais) [reply src/components/new-request/business-lookup.vue:93]: Kevin, please resolve this conversation when you have read it.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue:177]: Kevin, please resolve this conversation when you have read it.

File: src/components/new-request/business-lookup.vue
Comment (kzdev420) [reply src/components/new-request/business-lookup.vue:93]: I did

File: src/components/new-request/business-lookup.vue
Comment (kzdev420) [reply src/components/new-request/business-lookup.vue:98]: I did

File: src/components/new-request/search.vue
Comment (kzdev420) [reply src/components/new-request/search.vue:177]: I did

File: package.json
Comment (severinbeauvais) [inline package.json:15]: Check if "vue-debounce-decorator" (line 72) is still used by other components in this repo. If not then please uninstall it.

File: src/store/state.ts
Comment (severinbeauvais) [inline src/store/state.ts:162]: This looks too much like real data. Is there an "empty business lookup" const you could use here instead? Or could the whole object be set to null?

File: package.json
Comment (kzdev420) [reply package.json:15]: - There is one more that uses this. 
https://github.com/bcgov/namerequest/blob/ef3d33ccac9844200360816ad31c4e592aa4b1b1/src/mixins/action-mixin.ts#L130C1-L133

If we need to remove this too, I will do

File: src/store/state.ts
Comment (kzdev420) [reply src/store/state.ts:162]: I set to null

File: package.json
Comment (severinbeauvais) [reply package.json:15]: No, let's call this out of scope.

Resolving.

File: package.json
Comment (JazzarKarim) [reply package.json:1]: Kevin, just as Sev mentioned, please bump the version number on line 3 from `5.0.12` to `5.0.13` and do an `npm i`. That's also update the package.lock.json. Thanks!

File: src/components/new-request/search.vue
Comment (JazzarKarim) [inline src/components/new-request/search.vue:321]: Would it be possible to alphabetize these lines please? Also lines 21 and 22 in interfaces/index.ts. 

It's a small change but it'll keep the code clean for future readability for other devs. Thanks Kevin!

**Review by JazzarKarim — APPROVED** [review]
Aside from the small changes above, looks really good Kevin! 👍

Please wait for other approvals before merging 🙌

File: src/components/new-request/search.vue
Comment (kzdev420) [reply src/components/new-request/search.vue]: I did

Comment (JazzarKarim) [reply src/components/new-request/search.vue]: Awesome, thanks. Resolving.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:182]: I know that using business lookup is a work in progress, but could you make this conditional to CHG, AML, CNV and REH name requests only? That way it won't show for the other flows. Thanks.

File: src/components/new-request/search.vue
Comment (severinbeauvais) [inline src/components/new-request/search.vue:190]: Note to other devs working on Name Change flow, Amalgamation flow, Alteration flow or Reactivate flow -- delete this section and use the business name/ identifier/ legal type from the business search.

**Review by severinbeauvais — APPROVED** [review]
Good job, Kevin.

I think this PR is ready to merge, but like Karim is doing with Create UI, I think we should temporarily deploy this feature branch to Dev, test business lookup there, and then re-deploy main branch to Dev. This is to verify that the env keys are working correctly in Dev. LMK if you want to do this together.

File: package.json
Comment (severinbeauvais) [reply package.json:1]: Kevin, after updating the version, do a `npm i` and then commit both package.json and package-lock.json.

Comment (kzdev420) [reply package.json:1]: I did

File: src/components/new-request/search.vue
Comment (severinbeauvais) [reply src/components/new-request/search.vue:182]: ^^ to do in a future PR later today or tomorrow

**Pull request merged**
